nncf/tensorflow/quantization/algorithm.py

```diff
@@ -436,6 +499,25 @@ def _get_quantizer_setup(self, model: tf.keras.Model) -> TFQuantizationSetup:
 
         return setup
 
+    def _generate_unified_scale_groups(self,
+                                       model: tf.keras.Model,
+                                       quantizer_setup: SingleConfigQuantizerSetup,
+                                       qp_id_to_index: Dict[QuantizationPointId, int],
+                                       setup: TFQuantizationSetup):
+        # To properly set the instance indices for FQ need to save layers order like in the model config
+        layer_names = [layer.name for layer in model.layers]
+        for unified_group in quantizer_setup.unified_scale_groups.values():
+            sorted_unified_group = []
+            for qp_id in unified_group:
+                qp = quantizer_setup.quantization_points[qp_id]
+                qp_layer_name = qp.insertion_point.target_node_name
+                layer_idx = layer_names.index(qp_layer_name)
+                tf_setup_index = qp_id_to_index[qp_id]
+                sorted_unified_group.append((tf_setup_index, layer_idx))
+
+            sorted_unified_group = sorted(sorted_unified_group, key=lambda x: x[1])
+            setup.register_unified_scale_group([setup_index for setup_index, _ in sorted_unified_group])
```

Comment:
Since you are modifying the setup as a result of the function, return the setup explicitly instead of relying on Python's pass-by-reference.

Reply:
Done